ui/src/components/common/Holdings.tsx

```diff
@@ -99,36 +118,117 @@ const AllocationForm: React.FC<FormProps> = ({ asset, provider, role, depositCid
         setLoading(false);
     }
 
+    const handleMergeAssets = async (event: React.FormEvent) => {
+        event.preventDefault();
+        setLoading(true);
+
+        try {
+            const args = { depositCids: mergeAssets };
+            const cid = depositCid as ContractId<AssetDeposit>
+            await ledger.exercise(AssetDeposit.AssetDeposit_Merge, cid, args)
+            clearForm()
+
+        } catch (err) {
+            setError(parseError(err));
+        }
+        setLoading(false);
+    }
+
+    const handleSplitAsset = async (event: React.FormEvent) => {
+        event.preventDefault();
+        setLoading(true);
+
+        if (!splitAssetDecimal) {
+            return
+        }
+
+        if (splitAssetDecimal >= Number(asset.quantity)){
+            setLoading(false);
+            return setError({
+                header: 'Invalid Split Quantity',
+                message: `The splitting quantity must be less than ${asset.quantity}`
+            })
+        }
+
+        if (splitAssetDecimal <= 0 ){
+            setLoading(false);
+            return setError({
+                header: 'Invalid Split Quantity',
+                message: `The splitting quantity must be greater than 0.`
+            })
+        }
+
+        try {
+            const args = { quantities: [String(splitAssetDecimal)] };
```

Comment:
In an ideal world we would have to get a hold of the actual token, then fetch its `quantityPrecision` and either round or not allow the user to split in more decimals than the token suggests. Unless it is easy to add we can leave as is.